client/src/components/StepsQuestions/index.js

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import propTypes from 'prop-types';
+import { Steps, Button, message, Select, Radio } from 'antd';
+import { getCountryNames, getCities } from 'full-countries-cities';
+
+import UploadImg from './UploadImg';
+
+import './index.css';
+
+const { Step } = Steps;
+
+const handleChangeInput = (value, nameState, cb) => {
+  cb(value, nameState);
+};
+
+const renderInput = (values, current, funcs) => {
+  const stateKey = current === 0 ? 'name' : 'linkSite';
+  return (
+    <input
+      className="input "
+      type="text"
+      placeholder={current === 0 ? 'Type your answer here' : 'http://'}
+      value={current === 0 ? values.name : values.linkSite}
+      onChange={event =>
+        handleChangeInput(event.target.value, stateKey, funcs.handelChange)
+      }
+    />
+  );
+};
+
+const renderOptions = list => {
+  const { Option } = Select;
+
+  return list.map(item => (
+    <Option key={item} value={item}>
+      {item}
+    </Option>
+  ));
+};
+
+const dropDownFilter = (input, option) =>
+  option.props.children.toLowerCase().indexOf(input.toLowerCase()) >= 0;
+
+const renderSelect = (values, current, funcs) => {
+  const stateKey = current === 2 ? 'country' : 'city';
+  let cities;
+  if (current === 3) {
+    cities = values.country ? getCities(values.country) : [];
+  }
+  return (
+    <Select
+      className="ant-select-country"
+      showSearch
+      placeholder="Select"
+      optionFilterProp="children"
+      value={current === 2 ? values.country : values.city}
+      onChange={value => handleChangeInput(value, stateKey, funcs.handelChange)}
+      filterOption={dropDownFilter}
+    >
+      {current === 2 ? renderOptions(getCountryNames()) : renderOptions(cities)}
+    </Select>
+  );
+};
+
+const renderRadio = (values, funcs) => {
+  const BusinessTypes = Object.freeze({
+    stay: 'place to stay',
+    food: 'place to eat or drink',
+    shop: 'place to shop',
+  });
+
+  return (
+    <Radio.Group value={values.businessType} buttonStyle="solid">
+      {Object.entries(BusinessTypes).map(([key, value]) => (
+        <Radio.Button
+          key={key}
+          value={key}
+          onClick={event =>
+            handleChangeInput(
+              event.target.value,
+              'businessType',
+              funcs.handelChange
+            )
+          }
+        >
+          {value}
+        </Radio.Button>
+      ))}
+    </Radio.Group>
+  );
+};
+
+const renderQuestion = (questions, values, current, funcs, classes) => (
+  <div className={classes[current]}>
+    <h2 className={`title ${classes[current]}__title`}>
+      {questions[current].title}
+    </h2>
+    {questions[current].imgUrl && (
+      <div className="img__box">
+        <img
+          className="img__box--img"
+          src={questions[current].imgUrl}
+          alt="img question"
+        />
+      </div>
+    )}
+    {current < 2 && renderInput(values, current, funcs)}
```

Comment:
ok so (based on the fakeData.js) it looks that the questions will come from typeform? That means they are not part of the codebase, are configurable and can change. You can't be using their `id`s like this then! What if a new question will be inserted in the middle?

You will have to change this logic, instead of IDs, each question should contain some additional field like `inputType`, which you would use here to decide whether to render text field, radio button, etc.

Reply:
here we don't use typeform.

Reply:
wherever they come from, the questions are essentially data, they are not part of the code, correct? You cannot base your logic on their hardcoded IDs...